packages/admin/admin/src/form/FinalFormNumberInput.tsx

```diff
@@ -0,0 +1,84 @@
+import { InputBase, InputBaseProps } from "@mui/material";
+import * as React from "react";
+import { FieldRenderProps } from "react-final-form";
+import { useIntl } from "react-intl";
+
+import { ClearInputAdornment } from "../common/ClearInputAdornment";
+
+export type FinalFormNumberInputProps = InputBaseProps &
+    FieldRenderProps<number | string> & {
+        clearable?: boolean;
+        decimals?: number;
+    };
+
+export function FinalFormNumberInput({
+    meta,
+    input,
+    innerRef,
+    clearable,
+    endAdornment,
+    decimals = 0,
+    ...props
+}: FinalFormNumberInputProps): React.ReactElement {
+    const intl = useIntl();
+
+    const numberParts = intl.formatNumberToParts(1111.111);
+    const decimalSymbol = numberParts.find(({ type }) => type === "decimal")?.value;
+    const thousandSeparatorSymbol = numberParts.find(({ type }) => type === "group")?.value;
+
+    const [formattedNumberValue, setFormattedNumberValue] = React.useState("");
+
+    const getFormattedValue = React.useCallback(
+        (value, decimals: number) => {
+            const formattedValue =
+                value !== undefined ? intl.formatNumber(value, { minimumFractionDigits: decimals, maximumFractionDigits: decimals }) : "";
+            return formattedValue;
+        },
+        [intl],
+    );
+
+    const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        const { value } = event.target;
+        setFormattedNumberValue(value);
+    };
+
+    const handleBlur = (event: React.FocusEvent<HTMLInputElement>) => {
+        const { value } = event.target;
+
+        const numericValue = parseFloat(
+            value
+                .split(thousandSeparatorSymbol || "")
+                .join("")
+                .split(decimalSymbol || ".")
+                .join("."),
+        );
+
+        const inputValue = isNaN(numericValue) ? undefined : numericValue.toFixed(decimals);
```

Comment:
`toFixed()` returns a string, but the value should be a number. 
You can use `Math.round()`, when dividing then re-multiplying depending on the decimals. 

It could make sense to explicitly set the type before calling `onChange()` to prevent this kind of bug: 

```suggestion
        const inputValue: number | undefined = isNaN(numericValue) ? undefined : numericValue.toFixed(decimals);
```